ingestion/src/metadata/utils/dbt_config.py

```diff
@@ -0,0 +1,113 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Hosts the singledispatch to get DBT files
+"""
+import json
+import traceback
+import urllib.request
+from functools import singledispatch
+
+from metadata.generated.schema.metadataIngestion.databaseServiceMetadataPipeline import (
+    DbtGCSConfig,
+    DbtHttpConfig,
+    DbtLocalConfig,
+    DbtS3Config,
+)
+from metadata.ingestion.ometa.utils import ometa_logger
+from metadata.utils.credentials import set_google_credentials
+
+logger = ometa_logger()
+
+DBT_CATALOG_FILE_NAME = "catalog.json"
+DBT_MANIFEST_FILE_NAME = "manifest.json"
+
+
+@singledispatch
+def get_dbt_details(config):
+    try:
+        raise NotImplemented(
+            f"Config not implemented for type {type(config)}: {config}"
+        )
+    except Exception as exc:
```

Comment:
This except here is not really needed, as the try will always fail. Let's just keep the `NotImplementedError` (not `NotImplemented`) so that this blows up if something is not properly configured, otherwise we might be shadowing issues

Reply:
So the concept here to was to prevent the ingestion flow from breaking even if there was some issue with the DBT. This try except block will prevent the normal ingestion flow from breaking. 

Should I keep it that way or should I remove the try except block?

Reply:
I'll change the `NotImplemented` to `NotImplementedError`

Reply:
mmmh oki, let's now make it blow up but show the error in the usual logger, not the debug to make it clear 👌

Reply:
okay 👍